CMake/PackageConfig.cmake

```diff
@@ -1,9 +1,23 @@
 add_library(HighFive INTERFACE)
-target_link_libraries(HighFive INTERFACE ${HDF5_LIBRARIES})
+
+# Public headers
 target_include_directories(HighFive INTERFACE
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include>")
 target_include_directories(HighFive SYSTEM INTERFACE ${HDF5_INCLUDE_DIRS})
+target_link_libraries(HighFive INTERFACE ${HDF5_LIBRARIES})
+target_compile_definitions(HighFive INTERFACE ${HDF5_DEFINITIONS})
+
+# MPI
+if(HIGHFIVE_PARALLEL_HDF5)
+  target_include_directories(HighFive SYSTEM INTERFACE ${MPI_C_INCLUDE_PATH})
+  target_link_libraries(HighFive INTERFACE ${MPI_C_LIBRARIES})
+  if(CMAKE_VERSION VERSION_GREATER 3.13)
```

Comment:
Why this `if`? Can't MPI_C_LINK_FLAGS be always there?

Reply:
`target_link_options` is only available in recent version of cmake. And put this argument in `target_link_libraries` only work for gcc-like compiler.

Reply:
Lets try with this
```
set_target_properties(HighFive PROPERTIES LINK_FLAGS ${MPI_C_LINK_FLAGS})
```
It should do what is intended, even if not specifically INTERFACE. The default is normally PUBLIC which in this case is equivalent.

Reply:
Oups did it, but forget to push

Reply:
Oopps... we really need an IF
```
CMake Error at CMake/PackageConfig.cmake:15 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_FLAGS" is not allowed.

bash-5.0$ cmake --version
cmake version 3.14.4
```

Reply:
And no else?

Reply:
Like before, if someone complains we will think about making something work on all platform for cmake from 3.0 to 3.16+.
But maybe we can improve cmake version

Reply:
# In CMake 3.13+, LINK_FLAGS was converted to LINK_OPTIONS
See https://www.gitmemory.com/issue/LLNL/blt/266/503660028

Reply:
But no.
`VERSION_GREATER_EQUAL` does not exist on old cmake version